File: src/main/java/sds/offre/infra/IdFormuleGenerateurDeUUID.java
package sds.offre.infra;

import sds.offre.concept_metier.IdFormule;
import sds.offre.concept_metier.IdFormuleGenerateur;

import java.util.UUID;

public class IdFormuleGenerateurDeUUID implements IdFormuleGenerateur {
    public IdFormule nouveauId() {
        return IdFormule.de(UUID.randomUUID().toString());
    }
}

File: src/main/java/sds/offre/infra/IdFormuleGenerateurDeUUID.kt
package sds.offre.infra

import sds.offre.concept_metier.IdFormule
import sds.offre.concept_metier.IdFormuleGenerateur
import java.util.*

class IdFormuleGenerateurDeUUID : IdFormuleGenerateur {
    override fun nouveauId(): IdFormule {
        return IdFormule(UUID.randomUUID().toString())
    }
}
